### ISSUE TITLE
question about the result

### ISSUE THREAD

**Issue by yuzehui1996** [arangIN/SelfOrganizingMap-SOM#1]
I am confused about the loss did not decrease at all and the som_result.png was not the same as you showed.Should I change the hyperparameter such as lr or total epochs? Thanks a lot!

**Comment by Dotori-HJ** [comment]
Did you wait for the learning to end? During SOM learning, In my code change the hyperparameters to match the epoch. So you can see the desired results at the end of the learning.



If you want to see the results more quickly, reduce the number of trainning data.
You can insert below code, under 66 lines of <main.py>

```
train_data.train_data = train_data.train_data[:1000]
train_data.train_labels = train_data.train_labels[:1000]
```

That way, you'll only use 1,000 trainning data.
I hope you get a good result:)

**Comment by heitorrapela** [comment]
![Uploading som_30x30_all_data.gif…](Run all dataset mnist on SOM 30x30 grid)

**Comment by heitorrapela** [comment]
I used all dataset training with a 30x30 grid :+1: I got a good result:

![All Mnist Dataset trained with 30x30 SOM grid](https://media.giphy.com/media/SIJFViQbw0WnkVhHaA/giphy.gif)

**Comment by Bibyutatsu** [comment]
Hi, how can I get the mapping for my test set ?
It contains some images of handwriting and I want to know which cluster or location it belongs to.

**Comment by Dotori-HJ** [comment]
You can choose the closest one by comparing the weights of trained som with the value of your test data.

probably, pairwise distance will help you.
